dashboard/app/controllers/api/v1/pd/application/teacher_applications_controller.rb

```diff
@@ -13,14 +13,18 @@ def new_form
 
     # PATCH /api/v1/pd/application/teacher/<applicationId>
     def update
-      form_data_hash = params.try(:[], :form_data) || {}
-      form_data_json = form_data_hash.to_unsafe_h.to_json.strip_utf8mb4
+      form_data_hash = params.try(:[], :form_data)
+      form_data_json = form_data_hash.to_unsafe_h.to_json.strip_utf8mb4 if form_data_hash || '{}'
 
-      # [MEG] TODO: See if update is coming from hitting the submit button (do validations)
-      # or the save button (ignore validations)
       @application.form_data_hash = JSON.parse(form_data_json)
+      @application.set_status
+      @application.set_course_from_program
```

Comment:
why is this necessary if it will also be done `before_validation`?

Reply:
Good call. I was originally not doing `set_course_from_program` `before_validation` if program was nil, but that's not necessary now –– thanks!